DigitalLearningSolutions.Data/Services/TrackerActionService.cs

```diff
@@ -74,22 +74,41 @@ public TrackerEndpointResponse StoreDiagnosticJson(int? progressId, string? diag
                 return TrackerEndpointResponse.StoreDiagnosticScoreException;
             }
 
-            var diagnosticOutcomes = JsonConvert.DeserializeObject<IEnumerable<DiagnosticOutcome>>(diagnosticOutcome);
             try
             {
-                foreach (var diagOutcome in diagnosticOutcomes)
+                var diagnosticOutcomes = JsonConvert.DeserializeObject<IEnumerable<DiagnosticOutcome>>(
+                    diagnosticOutcome,
+                    new JsonSerializerSettings
+                    {
+                        MissingMemberHandling = MissingMemberHandling.Error,
+                        NullValueHandling = NullValueHandling.Include,
+                        DefaultValueHandling = DefaultValueHandling.Include,
+                    }
+                );
+
+                try

```

Comment:
I'm not certain we need the double `try-catch` as the calling method in the `TrackerService` will catch the json exception. 

But I get the reasoning though - to ensure that consistent output from this method.

Reply:
Yeah I wasn't sure if it mattered that the logged message in the catch statement wouldn't really reflect an issue going wrong in the deserialization. I've put it all in one try catch now.

Reply:
Yeah that makes sense. Given this try-catch does capture the exception then we can understand from logs why a failure may of occured.